apis/v1alpha1/httproute_types.go

```diff
@@ -535,26 +535,29 @@ type HTTPRouteForwardTo struct {
 	Port PortNumber `json:"port"`
 
 	// Weight specifies the proportion of traffic forwarded to the backend
-	// referenced by the ServiceName or BackendRef field. computed as
+	// referenced by the ServiceName or BackendRef field. This is computed as
 	// weight/(sum of all weights in this ForwardTo list). Weight is not a
-	// percentage and the sum of weights does not need to equal 100. If only one
-	// backend is specified, 100% of the traffic is forwarded to that backend.
-	// If weight is set to 0, no traffic should be forwarded for this entry.
-	// If unspecified, weight defaults to 1.
+	// percentage and the sum of weights does not need to equal 100.
+	//
+	// If only one backend is specified and it has a weight greater than 0, 100%
+	// of the traffic is forwarded to that backend. If weight is set to 0, no
+	// traffic should be forwarded for this entry. If unspecified, weight
+	// defaults to 1.
 	//
 	// Support: Core
 	//
 	// +kubebuilder:default=1
 	// +kubebuilder:validation:Minimum=0
-	// +kubebuilder:validation:Maximum=10000
+	// +kubebuilder:validation:Maximum=1000000000
```

Comment:
Not all implementations will be able to support arbitrary weights.
Kong, for example, has an upper limit of `65535`.

Reply:
Harry -- does it make sense to scale in the controller? will that work?
If it may be scaled, we should document.

Reply:
I think the key here is that this is calculated as `weight/(sum of all weights in this ForwardTo list)`. Not all/or many implementations will be able to provide this level of granularity, but they should be able to round this result into something the implementation can support. As far as documenting the scaling, are the current docs sufficient?

```
// Weight specifies the proportion of traffic forwarded to the backend
// referenced by the ServiceName or BackendRef field. This is computed as
// weight/(sum of all weights in this ForwardTo list).
```

On the other hand, I don't think this is a very important change, so if it ends up causing issues for some implementations I'm fine with pulling it.

Reply:
Couple of things:

* We should state proportion of HTTP requests to be specific
* Implementations can implement this in many different ways, so there may be some \epsilon from the exact proportion (w/ exception of the weight==0 case). This also means that `weight` should not be expected to be passed through w/out change unless the semantics of the underlying controller matches.

Reply:
👍 Updated spec to include that.

Reply:
Updated to 1 million - seemed like a less extreme upper limit but also low enough to ensure there will be no issues with a sum across multiple ForwardTo statements.